src/sinks/appsignal/tests.rs

```diff
@@ -0,0 +1,30 @@
+use futures::{future::ready, stream};
+
+use super::*;
+use crate::{
+    config::{SinkConfig as _, SinkContext},
+    event::{Event, LogEvent},
+    test_util::{
+        components::{run_and_assert_sink_compliance, HTTP_SINK_TAGS},
+        http::{always_200_response, spawn_blackhole_http_server},
+    },
+};
+
+#[test]
+fn generate_config() {
+    crate::test_util::test_generate_config::<AppsignalSinkConfig>();
+}
+
+#[tokio::test]
+async fn component_spec_compliance() {
+    let mock_endpoint = spawn_blackhole_http_server(always_200_response).await;
+
+    let config = format!("endpoint = \"{}\"\npush_api_key = \"abc\"", mock_endpoint);
+    let config: AppsignalSinkConfig = toml::from_str(&config).unwrap();
+
+    let context = SinkContext::new_test();
+    let (sink, _healthcheck) = config.build(context).await.unwrap();
+
+    let event = Event::Log(LogEvent::from("simple message"));
+    run_and_assert_sink_compliance(sink, stream::once(ready(event)), &HTTP_SINK_TAGS).await;
```

Comment:
I assume the HTTP_SINK_TAGS is the right choice here, since it sends it via HTTP?